packages/data-context/src/DataContext.ts

```diff
@@ -68,6 +68,10 @@ export class DataContext extends DataContextShell {
       this.actions.wizard.navigate('forward')
     }
 
+    if (this.config.launchArgs.browser) {
```

Comment:
As we put more logic in here, I start to wonder if this function is the wrong place for some of these things.

We make a `new DataContext`, then have a bunch of promises and other functions to mutate the state - it sort of feels like this should be done beforehand, so we can initialize the DataContext in the proper state to begin with? I'm not sure exactly where it should go, but a growing list of conditionals - mostly related to CLI and environmental args - starts to feel like this is in the wrong place.

Reply:
Yeah, I feel the same. Thoughts, @tgriesser?

Reply:
Maybe in another PR.

Reply:
Oops, thought I had commented on this yesterday - yeah we can fix in another PR, some of the work I'm doing to refactor to add the CLI arguments in open mode tests will indirectly help clean this up, and some of the other changes to restructure some of the GraphQL types & internal state in #18709 will also help clean this yp.